src/utils/compile.rs

```diff
@@ -0,0 +1,180 @@
+//! Utils to compile script to bytecode and run script from bytecode
+
+use libquickjs_sys as q;
+use crate::bindings::{OwnedValueRef, ContextWrapper};
+use crate::ExecutionError;
+use std::ffi::CString;
+
+/// compile a script, will result in a JSValueRef with tag JS_TAG_FUNCTION_BYTECODE or JS_TAG_MODULE.
+///  It can be executed with run_compiled_function().
+#[allow(dead_code)]
+pub fn compile<'a>(context: &'a ContextWrapper, script: &str, file_name: &str) -> Result<OwnedValueRef<'a>, ExecutionError> {
+    let filename_c = CString::new(file_name).map_err(|_e| { ExecutionError::Internal("cstring creation failed".to_string()) })?;
+    let code_c = CString::new(script).map_err(|_e| { ExecutionError::Internal("cstring creation failed".to_string()) })?;
+
+    let value_raw = unsafe {
+        q::JS_Eval(
+            context.context,
+            code_c.as_ptr(),
+            script.len() as _,
+            filename_c.as_ptr(),
+            q::JS_EVAL_FLAG_COMPILE_ONLY as i32,
+        )
+    };
+
+    // check for error
+    let ret = OwnedValueRef::new(
+        context,
+        value_raw,
+    );
+    if ret.is_exception() {
+        let ex_opt = context.get_exception();
+        if let Some(ex) = ex_opt {
+            Err(ex)
+        } else {
+            Err(ExecutionError::Internal("Unknown error".to_string()))
+        }
+    } else {
+        Ok(ret)
+    }
+}
+
+/// run a compiled function, see compile for an example
+#[allow(dead_code)]
+pub fn run_compiled_function<'a>(
+    context: &'a ContextWrapper,
+    compiled_func: &OwnedValueRef,
+) -> Result<OwnedValueRef<'a>, ExecutionError> {
+    assert!(compiled_func.is_compiled_function());
+    let val = unsafe { q::JS_EvalFunction(context.context, *compiled_func.into_inner()) };
+    let val_ref = OwnedValueRef::new(context, val);
+    if val_ref.is_exception() {
+        let ex_opt = context.get_exception();
+        if let Some(ex) = ex_opt {
+            Err(ex)
+        } else {
+            Err(ExecutionError::Internal(
+                "run_compiled_function failed and could not get exception".to_string()
+            ))
+        }
+    } else {
+        Ok(val_ref)
+    }
+}
+
+/// write a function to bytecode
+#[allow(dead_code)]
+pub fn to_bytecode(context: &ContextWrapper, compiled_func: &OwnedValueRef) -> Vec<u8> {
+    assert!(compiled_func.is_compiled_function());
+
+    #[cfg(target_pointer_width = "64")]
```

Comment:
Just use `usize` here. `usize` is `u32` on 32bit systems and `u64` on 64 bit.

Reply:
It's also possible to just use `as _` and let the type system figure it out.

Reply:
> It's also possible to just use `as _` and then letting the type system figure it out.
Thanks for the tip!

That works for the passing len for one case, when using it as the return val 
it gives me
```
error[E0606]: casting `&mut i32` as `*mut u64` is invalid
  --> src/utils/compile.rs:79:13
   |
79 |             &mut len as _,
   |             ^^^^^^^^^^^^^

error: aborting due to previous error
```